### PR TITLE
feat(chat): Improve responsiveness.

### PR DESCRIPTION
**Pull request by muscat1**
* Fix toolbox buttons not displaying properly when chat is open.
* Open chat in fullscreen dialog past custom thresholds when mobile/desktop toolbox would become unusable due to chat
* Remove mobile chat check when displaying toolbox

### CODE REVIEW

**Comment by muscat1** [comment]
Took some screenshots to exemplify how the screen would look when opening chat:

Desktop browser - enter nickname:

Screenshot 2020-12-16 at 12 46 02

Desktop browser - chat example (close to max width):

Screenshot 2020-12-16 at 13 15 48

Mobile browser - chat example (close to max width):

Screenshot 2020-12-16 at 13 16 37

File: react/features/chat/components/AbstractChat.js [hunk 1]
```diff
@@ -106,6 +113,9 @@ export function _mapStateToProps(state: Object) {
     const _localParticipant = getLocalParticipant(state);
 
     return {
+        _isModal: isMobileBrowser()
+            ? window.innerWidth <= MOBILE_SMALL_WIDTH_THRESHOLD
```
Comment (saghul) [inline react/features/chat/components/AbstractChat.js:117]: Does this work when resizing dynamically? I wonder if these props will be computed at the right time.

Comment (muscat1) [reply react/features/chat/components/AbstractChat.js:117]: Yup, dragging past the thresholds switches between the sidebar view and the modal one

**Comment by muscat1** [comment]
Jenkins please test this please

File: css/_chat.scss [hunk 2]
```diff
@@ -379,3 +379,30 @@
...
+        justify-content: space-between;
+        margin: 16px 16px 24px;
+        width: calc(100% - 32px);
+        box-sizing: border-box;
+        color: #fff;
+        font-weight: 600;
+        font-size: 24px;
+        line-height: 32px;
+
+        & > div > svg {
```
[the remark this answers is not in the document]
Comment (muscat1) [reply css/_chat.scss]: Done!

File: css/_responsive.scss [hunk 3]
```diff
@@ -65,3 +65,171 @@
...
+            width: 100%;
+        }
+
+        &.shift-right [class^="Modal__Dialog-"] {
+            height: 100%;
+            max-height: 100%;
+            border-radius: 0;
+        }
+    }
+}
```
[the remark this answers is not in the document]
Comment (muscat1) [reply css/_responsive.scss:199]: Added some mixins (for the above styles as well)

**Review by saghul — APPROVED** [review]
Excellent job Mihai! 🚀 🍰 🎨

**Pull request merged**